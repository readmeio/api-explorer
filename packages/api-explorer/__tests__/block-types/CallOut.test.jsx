const React = require('react');
const { shallow, mount } = require('enzyme');
const CallOut = require('../../src/block-types/CallOut');

test('should render title', () => {
  const block = {
    data: {
      type: 'info',
      title: 'Callout',
    },
  };
  const callout = mount(<CallOut block={block} />);
  expect(callout.find('h3').text()).toBe('Callout');
});

describe('icons', () => {
  const icons = {
    info: 'info-circle',
    warning: 'exclamation-circle',
    danger: 'exclamation-triangle',
    success: 'check-square',
  };

  it('should render with title', () => {
    Object.keys(icons).forEach(type => {
      const className = icons[type];
      const block = {
        data: {
          type,
          title: 'Callout',
        },
      };
      expect(mount(<CallOut block={block} />).find(`.fa-${className}`)).toHaveLength(1);
    });
  });

  it('should render without title', () => {
    Object.keys(icons).forEach(type => {
      const className = icons[type];
      const block = {
        data: {
          type,
        },
      };
      expect(mount(<CallOut block={block} />).find(`.noTitleIcon .fa-${className}`)).toHaveLength(1);
    });
  });
});

test('should render nothing if no title and icon', () => {
  const block = {
    data: {
      type: '',
    },
  };
  const callout = mount(<CallOut block={block} />);
  expect(callout.find('span').text()).toBe('');
});

test('should render body', () => {
  const block = {
    data: {
      type: 'info',
      body: 'body',
    },
  };

  const callout = shallow(<CallOut block={block} />);
  expect(
    callout
      .render()
      .find('.callout-body')
      .html()
  ).toMatchSnapshot();
});

test('should render markdown in body', () => {
  const block = {
    data: {
      type: 'info',
      body: '# heading\n`test`',
    },
  };

  const callout = shallow(<CallOut block={block} />);
  expect(
    callout
      .render()
      .find('.callout-body')
      .html()
  ).toMatchSnapshot();
});
