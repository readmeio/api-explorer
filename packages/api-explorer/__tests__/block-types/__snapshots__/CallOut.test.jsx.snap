// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render body 1`] = `"<p>body</p>"`;

exports[`should render markdown in body 1`] = `
"<h1 class=\\"header-scroll\\"><div class=\\"anchor waypoint\\" id=\\"section-heading\\"></div>heading<a class=\\"fa fa-anchor\\" href=\\"#section-heading\\"></a></h1>
<p><code><span class=\\"cm-s-neo\\">test</span></code></p>"
`;
